app/modules/net.c

```diff
@@ -883,25 +888,24 @@ static int net_dns_static( lua_State* L ) {
     return luaL_error(L, "wrong domain");
   }
 
+  /* Register callback with registry */
   luaL_checkanyfunction(L, 2);
-  lua_pushvalue(L, 2);  // copy argument (func) to the top of stack
+  lua_pushvalue(L, 2);
   int cbref = luaL_ref(L, LUA_REGISTRYINDEX);
-  if (cbref == LUA_NOREF) {
-    return luaL_error(L, "wrong callback");
-  }
-  int *cbref_ptr = calloc(1, sizeof(int));
-  cbref_ptr[0] = cbref;
   ip_addr_t addr;
-  err_t err = dns_gethostbyname(domain, &addr, net_dns_static_cb, cbref_ptr);
+
+  _Static_assert(sizeof(void *) >= sizeof(int), "Can't upcast int to ptr");
+
```

Comment:
Is it possible to use siomething like sizof(typeof(cbref)) to be save if cbref changes its type (to long) But maybe that's a bit too unlikely.

Reply:
Sure.